src/corelibs/U2Private/src/TaskSchedulerImpl.cpp

```diff
@@ -147,7 +147,7 @@ bool TaskSchedulerImpl::processFinishedTasks() {
         TaskInfo* ti = priorityQueue[i];
         TaskInfo* pti = ti->parentTaskInfo;
         Task::State state = ti->task->getState();
-        U2_ASSERT(state != Task::State_Finished);
+        SAFE_POINT(state != Task::State_Finished, "Task state is not 'finished'.", hasFinished);
```

Comment:
Add task details here, so the error message will be more informative (same in all other places)